operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/util/BulkDataImportUtil.java

```diff
@@ -31,66 +31,79 @@
  * BulkData Import Util captures common methods
  */
 public class BulkDataImportUtil {
-    private BulkDataImportUtil() {
-        // No Operation
-    }
+    private FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
+    private EvaluationContext evaluationContext = null;
 
-    public static FHIROperationException buildExceptionWithIssue(String msg, IssueType issueType)
-            throws FHIROperationException {
-        OperationOutcome.Issue ooi = FHIRUtil.buildOperationOutcomeIssue(msg, issueType);
-        return new FHIROperationException(msg).withIssue(ooi);
-    }
+    public BulkDataImportUtil(Parameters parameters) throws FHIROperationException {
+        if (parameters == null) {
+            throw buildExceptionWithIssue("$import parameters are empty or null", IssueType.INVALID);
+        }
 
-    public static FHIROperationException buildExceptionWithIssue(String msg, Throwable cause, IssueType issueType)
-            throws FHIROperationException {
-        OperationOutcome.Issue ooi = FHIRUtil.buildOperationOutcomeIssue(msg, issueType);
-        return new FHIROperationException(msg, cause).withIssue(ooi);
+        evaluationContext = new EvaluationContext(parameters);
     }
 
-    public static String retrieveInputFormat(Parameters parameters) throws FHIROperationException {
-        // Parameter: inputFormat (required)
-        // If there are multiple entries, the processing only takes the first entry.
-        if (parameters != null) {
-            for (Parameters.Parameter parameter : parameters.getParameter()) {
-                if (BulkDataConstants.PARAM_INPUT_FORMAT.equals(parameter.getName().getValue())
-                        && parameter.getValue() != null
-                        && parameter.getValue().is(com.ibm.fhir.model.type.String.class)) {
-                    // If the parameter isn't passed, use application/fhir+ndjson
-                    // Check the MediaType
-                    String val = parameter.getValue().as(com.ibm.fhir.model.type.String.class).getValue();
-                    if (BulkDataConstants.INPUT_FORMATS.contains(val)) {
-                        return val;
-                    }
+    /**
+     * processes the parameter inputFormat from the Parameters object and evaluationContext.
+     *
+     * @implNote If there are multiple entries, the processing only takes the first entry that matches.
+     *
+     * @return
+     * @throws FHIROperationException
+     */
+    public String retrieveInputFormat() throws FHIROperationException {
+        try {
+            Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, "parameter.where(name = 'inputFormat').value");
+            Iterator<FHIRPathNode> iter = result.iterator();
+            while (iter.hasNext()) {
+                FHIRPathElementNode node = (FHIRPathElementNode) iter.next();
```

Comment:
really no different, but since we have it:
```suggestion
                FHIRPathElementNode node = iter.next().as(FHIRPathElementNode.class);
```